.githooks/pre-commit

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+#
+# hook script to verify code formatting and runs tests for all modules
+# also it will try to format code and add missing license headers to every kotlin file.
+
+echo "***** Run code formatting and tests ******"
+git stash -q --keep-index
+
+./gradlew check
```

Comment:
shouldn't we also run `spotlessApply` or `spotlessCheck`?

What does `check` task run?

Reply:
I added `spotlessApply` to the script.
`test` is running `build`, `lint`, `test` and `spotlessCheck` already